src/models/app_data_json_format.rs

```diff
@@ -116,14 +133,130 @@ mod tests {
+                "metadata":{
+                    "environment": "production",
+                    "referrer":{
+                        "kind":"referrer",
+                        "address":"0x8c35B7eE520277D14af5F6098835A584C337311b",
+                        "version":"6.6.6"
+                },
+                "quote": {
+                    "version": "2.0",
+                    "slippageBips": "5"
+                }
+            }
+        });
+        let json: AppData = serde_json::from_value(value).unwrap();
+        let expected = AppData {
+            version: "1.2.3".to_string(),
+            app_code: "MooSwap".to_string(),
+            environment: Some("production".to_string()),
+            metadata: Some(Metadata {
+                environment: Some("production".to_string()),
+                referrer: Some(Referrer {
+                    address: "0x8c35B7eE520277D14af5F6098835A584C337311b"
+                        .parse()
+                        .unwrap(),
+                    version: "6.6.6".to_string(),
+                }),
+                quote: Some(Quote::V2(QuoteV2 {
+                    version: String::from("2.0"),
+                    slippage_bips: 5,
+                })),
+            }),
+        };
+
+        assert_eq!(json, expected);
+    }
+    #[test]
+    fn test_loading_json_v4_with_amounts() {
+        let value = json!({
+                "version":"1.2.3",
+                "appCode":"MooSwap",
+                "environment": "production",
+                "metadata":{
+                    "environment": "production",
+                    "referrer":{
+                        "kind":"referrer",
+                        "address":"0x8c35B7eE520277D14af5F6098835A584C337311b",
+                        "version":"6.6.6"
+                },
+                "quote": {
+                    "version": "2.0",
+                    "buyAmount": "123456789",
+                    "sellAmount": "123456789101112",
+                }
+            }
+        });
+        let json: AppData = serde_json::from_value(value).unwrap();
+        let expected = AppData {
+            version: "1.2.3".to_string(),
+            app_code: "MooSwap".to_string(),
+            environment: Some("production".to_string()),
+            metadata: Some(Metadata {
+                environment: Some("production".to_string()),
+                referrer: Some(Referrer {
+                    address: "0x8c35B7eE520277D14af5F6098835A584C337311b"
+                        .parse()
+                        .unwrap(),
+                    version: "6.6.6".to_string(),
                 }),
+                // Notice that although we have specified v2 for quote, we fall back on the v1 quote.
+                // This is much cleaner than trying to implement optional fields for each of
+                // the three components and validate that
+                // only one of slippage OR (both of sell/buy amounts are populated)
```

Comment:
@alfetopito i think this comment will help with your question above. 

You can look above at the implementation of quote and see how v1 and v2 are specified. This means even if you pass v2 with the v1 spec, it will just be parsed as a the right kind of quote. 

Also, the fact that metadata didn't need to be touched at all indicates why it didn't need a version bump.